src/main/kotlin/net/atos/zac/app/planitems/PlanItemsRESTService.kt

```diff
@@ -0,0 +1,301 @@
+            humanTaskData.toelichting,
+            taakdata,
+            zaakUUID
+        )
+        indexingService.addOrUpdateZaak(zaakUUID, false)
+    }
+
+    @POST
+    @Path("doProcessTaskPlanItem")
+    fun doProcessTaskplanItem(processTaskData: RESTProcessTaskData) =
+        cmmnService.startProcessTaskPlanItem(processTaskData.planItemInstanceId, processTaskData.data)
+
+    @POST
+    @Path("doUserEventListenerPlanItem")
+    fun doUserEventListenerPlanItem(userEventListenerData: RESTUserEventListenerData) {
+        val zaak = zrcClientService.readZaak(userEventListenerData.zaakUuid)
+        val zaakRechten = policyService.readZaakRechten(zaak)
+        PolicyService.assertPolicy(zaakRechten.startenTaak)
+        if (userEventListenerData.restMailGegevens != null) {
+            PolicyService.assertPolicy(zaakRechten.versturenEmail)
+        }
+        when (userEventListenerData.actie) {
+            UserEventListenerActie.INTAKE_AFRONDEN -> {
+                val planItemInstance = cmmnService.readOpenPlanItem(
+                    userEventListenerData.planItemInstanceId
+                )
+                zaakVariabelenService.setOntvankelijk(planItemInstance, userEventListenerData.zaakOntvankelijk)
+                if (!userEventListenerData.zaakOntvankelijk) {
+                    policyService.checkZaakAfsluitbaar(zaak)
+                    val zaakafhandelParameters = zaakafhandelParameterService.readZaakafhandelParameters(
+                        UriUtil.uuidFromURI(zaak.zaaktype)
+                    )
+                    zgwApiService.createResultaatForZaak(
+                        zaak,
+                        zaakafhandelParameters.nietOntvankelijkResultaattype,
+                        userEventListenerData.resultaatToelichting
+                    )
+                }
+            }
+
+            UserEventListenerActie.ZAAK_AFHANDELEN -> {
+                policyService.checkZaakAfsluitbaar(zaak)
+                if (!brcClientService.listBesluiten(zaak).isEmpty()) {
+                    val resultaat = zrcClientService.readResultaat(zaak.resultaat)
+                    resultaat.toelichting = userEventListenerData.resultaatToelichting
+                    zrcClientService.updateResultaat(resultaat)
+                } else {
+                    zgwApiService.createResultaatForZaak(
+                        zaak,
+                        userEventListenerData.resultaattypeUuid,
+                        userEventListenerData.resultaatToelichting
+                    )
+                }
+            }
+        }
+        cmmnService.startUserEventListenerPlanItem(userEventListenerData.planItemInstanceId)
+        if (userEventListenerData.restMailGegevens != null) {
+            mailService.sendMail(
+                restMailGegevensConverter.convert(userEventListenerData.restMailGegevens),
+                zaak.getBronnenFromZaak()
+            )
+        }
+    }
+
+    @Suppress("ReturnCount")
```

Comment:
Yeah, not very nice those 3 return statements I feel. But oh well, for another day. :-)

Reply:
Extracted the code in a new function and now we have only 2 returns, which makes detekt happy